frontend/public/components/secrets/create-secret.tsx

```diff
@@ -153,50 +154,86 @@ const withSecretForm = (SubForm) => class SecretFormComponent extends React.Comp
+                  value={this.state.secret.metadata.name}
+                  aria-describedby="secret-name-help"
+                  id="secret-name"
+                  required
+                />
+                <p className="help-block" id="secret-name-help">Unique name of the new secret.</p>
+              </div>
             </div>
-          </div>
-        </fieldset>
-        <SubForm
-          onChange={this.onDataChanged}
-          onError={this.onError}
-          onFormDisable={this.onFormDisable}
-          stringData={this.state.stringData}
-          secretType={this.state.secret.type}
-          isCreate={this.props.isCreate}
-        />
-        <ButtonBar errorMessage={this.state.error} inProgress={this.state.inProgress}>
-          <button type="submit" disabled={this.state.disableForm} className="btn btn-primary" id="save-changes">{this.props.saveButtonText || 'Create'}</button>
-          <button type="button" className="btn btn-default" id="cancel" onClick={onCancel}>Cancel</button>
-        </ButtonBar>
-      </form>
-    </div>;
+          </fieldset>
+          <SubForm
+            onChange={this.onDataChanged}
+            onError={this.onError}
+            onFormDisable={this.onFormDisable}
+            stringData={this.state.stringData}
+            secretType={this.state.secret.type}
+            isCreate={this.props.isCreate}
+          />
+          <ModalSubmitFooter errorMessage={this.state.error || ''} inProgress={this.state.inProgress} submitText="Create" cancel={this.props.onCancel} />
+        </ModalBody>
+      </form>:
+      (<div className="co-m-pane__body">
+        <Helmet>
+          <title>{title}</title>
+        </Helmet>
+        <form className="co-m-pane__body-group co-create-secret-form co-m-pane__form" onSubmit={this.save}>
+          <h1 className="co-m-pane__heading">{title}</h1>
+          <p className="co-m-pane__explanation">{this.props.explanation}</p>
+          <fieldset disabled={!this.props.isCreate}>
+            <div className="form-group">
+              <label className="control-label co-required" htmlFor="secret-name">Secret Name</label>
+              <div>
+                <input className="form-control"
+                  type="text"
+                  onChange={this.onNameChanged}
+                  value={this.state.secret.metadata.name}
+                  aria-describedby="secret-name-help"
+                  id="secret-name"
+                  required
+                />
+                <p className="help-block" id="secret-name-help">Unique name of the new secret.</p>
+              </div>
+            </div>
+          </fieldset>
+          <SubForm
+            onChange={this.onDataChanged}
+            onError={this.onError}
+            onFormDisable={this.onFormDisable}
+            stringData={this.state.stringData}
+            secretType={this.state.secret.type}
+            isCreate={this.props.isCreate}
+          />
```

Comment:
I see there is a lot of duplication of code going on here. I will suggest you create a new component that can be used in both cases where you want to show it as a form page/ as a modal. That way the file size will also be reduced. 

Next time when you have implemented something in a certain way just try and review your code and see if there is a lot of code duplication, if yes then try to minimize it. Also, try to make reusable components wherever possible.